booster_nrpe/booster_nrpe.py

```diff
@@ -90,53 +102,19 @@ def build_query(self, command):
         crc = 0
 
         if not command:
-            self.state = 'received'
             self.rc = 3
             self.message = "Error : no command asked from nrpe query"
+            self.state = 'received'
             return
 
         # We pack it, then we compute CRC32 of this first query
-        self.query = struct.pack(">2hih1024scc", 02, 01, crc, 0, command, 'N', 'D')
-        crc = binascii.crc32(self.query)
+        query = struct.pack(">2hih1024scc", 02, 01, crc, 0, command, 'N', 'D')
+        crc = binascii.crc32(query)
 
         # we restart with the crc value this time
         # because python2.4 do not have pack_into.
         self.query = struct.pack(">2hih1024scc", 02, 01, crc, 0, command, 'N', 'D')
```

Comment:
the query packet first 2 bytes are for the protocol version used effectively. which is here hardcoded to 2 because this module only speak that version of the protocol.
no more no less :s